File: leaders.py
```python
def leaders(arr):
    max_seen = arr[-1]
    leaders_list = [max_seen]
    for num in arr[:-1][::-1]:
        if num >= max_seen:
            max_seen = num
            leaders_list.append(max_seen)
    return leaders_list[::-1]

File: leaders.cpp
#include <vector>
using namespace std;

vector<int> leaders(vector<int>& arr) {
    vector<int> result;
    for (int i = arr.size() - 1; i >= 0; i--) {
        bool leader = true;
        for (int j = i + 1; j < arr.size(); j++) {
            if (arr[j] >= arr[i]) {
                leader = false;
                break;
            }
        }
        if (leader) {
            result.push_back(arr[i]);
        }
    }
    return result;
}